test/ShallowWrapper-spec.jsx

```diff
@@ -2777,13 +2777,27 @@ describe('shallow', () => {
         ]);
       });
 
-      it('calls expected methods when receiving new context', () => {
-        wrapper.setContext({ foo: 'foo' });
-        expect(spy.args).to.deep.equal([
-          ['shouldComponentUpdate'],
-          ['componentWillUpdate'],
-          ['render'],
-        ]);
+      describeIf(REACT013 || REACT15, 'setContext', () => {
+        it('calls expected methods when receiving new context', () => {
+          wrapper.setContext({ foo: 'foo' });
+          expect(spy.args).to.deep.equal([
+            ['componentWillReceiveProps'],
+            ['shouldComponentUpdate'],
+            ['componentWillUpdate'],
+            ['render'],
+          ]);
+        });
+      });
+
+      describeIf(REACT014, 'setContext', () => {
```

Comment:
In React 13 and 14, the check for whether willReceiveProps is called is just a strict equality check in ReactCompositeComponent between the previous and next elements. In React 15, they added an additional equality check between previous and next context.

willReceiveProps is called in 13 because Enzyme clones the element on each render in createRendererCompatible, failing the equality check.

It's not called in 14 because the nodes are equal.

It's called in 15 because of the new check for context.